components/add_user_to_channel_modal/add_user_to_channel_modal.jsx

```diff
@@ -0,0 +1,316 @@
+import SuggestionList from 'components/suggestion/suggestion_list.jsx';
+
+import {placeCaretAtEnd} from 'utils/utils.jsx';
+
+export default class AddUserToChannelModal extends React.Component {
+    static propTypes = {
+
+        /**
+         * Function that's called when modal is closed
+         */
+        onHide: PropTypes.func.isRequired,
+
+        /**
+         * The user that is being added to a channel
+         */
+        user: PropTypes.object.isRequired,
+
+        /**
+         * Object used to determine if the user
+         * is a member of a given channel
+         */
+        channelMembers: PropTypes.object.isRequired,
+
+        actions: PropTypes.shape({
+
+            /**
+             * Function to add the user to a channel
+             */
+            addChannelMember: PropTypes.func.isRequired,
+
+            /**
+             * Function to fetch the user's channel membership
+             */
+            getChannelMember: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+
+            /**
+             * Whether or not the modal is visible
+             */
+            show: true,
+
+            /**
+             * Whether or not a request to add the user is in progress
+             */
+            saving: false,
+
+            /**
+             * Whether or not a request to check for the user's channel membership
+             * is in progress
+             */
+            checkingForMembership: false,
+
+            /**
+             * The user input in the channel search box
+             */
+            text: '',
+
+            /**
+             * The id for the channel that is selected
+             */
+            selectedChannelId: null,
+
+            /**
+             * An error to display when the add request fails
+             */
+            inviteError: '',
+        };
+        this.suggestionProviders = [new SearchChannelWithPermissionsProvider()];
+        this.enableChannelProvider();
+    }
+
+    enableChannelProvider = () => {
+        this.suggestionProviders[0].disableDispatches = false;
+    }
+
+    focusTextbox = () => {
+        if (this.channelSearchBox == null) {
+            return;
+        }
+
+        const textbox = this.channelSearchBox.getTextbox();
+        if (document.activeElement !== textbox) {
+            textbox.focus();
+            placeCaretAtEnd(textbox);
+        }
+    }
+
+    onChange = (e) => {
+        this.setState({text: e.target.value, selectedChannelId: null});
+    }
+
+    onHide = () => {
+        this.setState({show: false});
+        this.props.onHide();
+    }
+
+    setSearchBoxRef = (input) => {
+        this.channelSearchBox = input;
+        this.focusTextbox();
+    }
+
+    handleInviteError = (error) => {
+        if (error) {
+            this.setState({inviteError: error.message, saving: false});
```

Comment:
`inviteError` seems like a slightly odd name for the property here (and the method name).

Reply:
Right, the action here is **adding** the user to the channel, not **inviting** the user. This method is only called when there is an error upon submitting the form, so I'll change to `submitError` and `handleSubmitError()`.

I'm also changing the local variable for displaying the error in the `render()` method to `errorMsg`, since `submitError` doesn't necessarily apply in all cases of showing an error message here.